tls/client-tls13-resume.c

```diff
@@ -228,7 +228,22 @@ int main(int argc, char** argv)
     wolfSSL_FreeArrays(ssl);
 #endif
 
-    /* Save the session */
+    /* Save the session
+     *
+     * NOTE: This is not an ideal solution. Please see client-tls-resume.c.
+     * There you will see that we do not need to call wolfSSL_peek() because
+     * we have already called wolfSSL_read() before calling
+     * wolfSSL_get_session() so wolfSSL has already internally stored the
+     * session ticket. For TLS 1.2 it is fine to put wolfSSL_get_session()
+     * right after the connection is established because the resumption
+     * information is part of the TLS 1.2 handshake. This is not the case for
+     * TLS 1.3. However, if you are migrating from TLS 1.2 and are having a
+     * hard time moving your call to wolfSSL_get_session(), you can try this
+     * approach.
+     *
+     * This approach can result in issues with I/O and is best used with non-
+     * blocking mode sockets.
+     */
```

Comment:
This comment should also indicate the `wolfSSL_get_session` below prior to shutdown is implemented below and is the recommended solution. Also feel free to `#if 0` this block of code.

Reply:
Good idea!!